server/src/main/java/com/flipkart/varadhi/web/AuthorizationHandlerBuilder.java

```diff
@@ -23,18 +25,28 @@
 public class AuthorizationHandlerBuilder {
 
     private final AuthorizationProvider provider;
+    private final Timer timer;
 
-    public AuthorizationHandlerBuilder(AuthorizationProvider provider) {
+    private final MeterRegistry meterRegistry;
+
+    public AuthorizationHandlerBuilder(AuthorizationProvider provider, MeterRegistry meterRegistry) {
         this.provider = Objects.requireNonNull(provider, "Authorization Provider is null");
+        this.meterRegistry = Objects.requireNonNull(meterRegistry, "Meter registry is null");
```

Comment:
if you have already registered the timer, then you porbably dont need to meterRegistry saeved as class variable.

Reply:
this is needed to start the clock. 

Timer.start(MeterRegistry meterRegistry) -> uses clock used by registered meterRegistry
Timer.start() -> uses system clock. 
Timer.start(Clock clock) -> uses explicitly defined clock.